src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsworkload/service/staff/JpaBasedGetEventManager.kt

```diff
@@ -1,11 +1,21 @@
 package uk.gov.justice.digital.hmpps.hmppsworkload.service.staff
 
+import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.EventCase
 import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.EventManagerEntity
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.repository.CaseDetailsRepository
 import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.repository.EventManagerRepository
 import java.util.UUID
 
 @Service
-class JpaBasedGetEventManager(private val eventManagerRepository: EventManagerRepository) : GetEventManager {
+class JpaBasedGetEventManager(private val eventManagerRepository: EventManagerRepository, private val caseDetailsRepository: CaseDetailsRepository) : GetEventManager {
   override fun findById(id: UUID): EventManagerEntity? = eventManagerRepository.findByUuid(id)
+  override fun findLatestByStaffAndTeam(staffCode: String, teamCode: String): EventCase? = eventManagerRepository.findByStaffCodeAndTeamCodeLatest(staffCode, teamCode)
+    .filter { caseDetailsRepository.existsById(it.crn) }
```

Comment:
query for exists per event manager entry, can't see an elegant way of making this 1 query for all event manager entries returned